command.c

```diff
@@ -274,28 +274,29 @@ void redis_parse_cmd(struct cmd *r) {
             goto error;
         }
 
-        /* Skip forward to the 'startfrom' arg index. */
+        /* Skip forward to the 'startfrom' arg index, then search for the keyword. */
         arg = arg1;
         arglen = arg1_len;
-        while (argidx < startfrom) {
+        while (argidx < (int)rnarg - 1) {
             if ((p = redis_parse_bulk(p, end, &arg, &arglen)) == NULL)
                 goto error; /* Keyword not provided, thus no keys. */
-            argidx++;
+            if (argidx++ < startfrom)
+                continue; /* Keyword can't appear in a position before 'startfrom' */
+            if (!strncasecmp(keyword, arg, arglen)) {
+                /* Keyword found. Now the first key is the next arg. */
```

Comment:
@zyhuf We need to skip all arguments before 'startfrom', like this. Otherwise we have a problem if the group name or the consumer name is 'stream'.

Reply:
I ran the modified code locally, and the test still showed an error, as indicated below.
“Failed to find keys of command”

Then, after I removed the reference to 'startfrom,' it ran successfully. @zuiderkwast

Reply:
I figured out the reason, thank you for the explanation. @zuiderkwast